assets/app/scripts/directives/oscKeyValues.js

```diff
@@ -22,6 +22,9 @@ angular.module("openshiftConsole")
   .controller("KeyValuesController", function($scope){
     var added = {};
     $scope.allowDelete = function(value){
+      if ($scope.canBeEmpty && (Object.keys($scope.entries).length === 1)) {
+        return false;
+      }
```

Comment:
Maybe I have this backwards, but I would expect delete to be allowed if `canBeEmpty` is true.